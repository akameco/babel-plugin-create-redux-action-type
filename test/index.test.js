// @flow
import { transform } from 'babel-core'
import plugin from '../src'

function transformCode(input /* : string */, opts /* : ?Object */ = {}) {
  const { code } = transform(input, {
    plugins: [[plugin, opts]],
  })
  return code
}

test('add flow comments', () => {
  const input = `
    type HOGE_TYPE = 'hoge/HOGE_ACTION'
  `
  expect(transformCode(input)).toMatchSnapshot()
})

test('add flow comments when exits flow comments', () => {
  const input = `
    // @flow
    type HOGE_TYPE = 'hoge/HOGE_ACTION'
    type FUGA_TYPE = 'fuga/FUGA_ACTION'
  `
  expect(transformCode(input)).toMatchSnapshot()
})

test('add flow comments when exits flow block comments', () => {
  const input = `
    /* @flow */
    type HOGE_TYPE = 'hoge/HOGE_ACTION'
    type FUGA_TYPE = 'fuga/FUGA_ACTION'
  `
  expect(transformCode(input)).toMatchSnapshot()
})

test('with extra', () => {
  const input = `
    /* @flow */
    type HOGE_TYPE = 'hoge/HOGE_ACTION'
    type FUGA_TYPE = 'fuga/FUGA_ACTION'

    const hoge = 'hoge'
    const fuga = 'fuga'

    type Action = { +type: HOGE_TYPE | HUGA_TYPE }
  `
  expect(transformCode(input)).toMatchSnapshot()
})

test('opts.filename', () => {
  const input = `
    /* @flow */
    type HOGE_TYPE = 'hoge/HOGE_ACTION'
    type FUGA_TYPE = 'fuga/FUGA_ACTION'

    const hoge = 'hoge'
    const fuga = 'fuga'

    type Action = { +type: HOGE_TYPE | HUGA_TYPE }
  `
  expect(transformCode(input, { filename: 'actionTypes' })).toMatchSnapshot()
})
