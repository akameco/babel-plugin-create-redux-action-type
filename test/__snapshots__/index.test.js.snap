// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add flow comments 1`] = `
"// @flow
import type { HOGE_TYPE } from './unknown';

export const HOGE: HOGE_TYPE = 'hoge/HOGE_ACTION';"
`;

exports[`add flow comments when exits flow block comments 1`] = `
"// @flow
import type { HOGE_TYPE, FUGA_TYPE } from './unknown';

export const HOGE: HOGE_TYPE = 'hoge/HOGE_ACTION';
export const FUGA: FUGA_TYPE = 'fuga/FUGA_ACTION';"
`;

exports[`add flow comments when exits flow comments 1`] = `
"// @flow
import type { HOGE_TYPE, FUGA_TYPE } from './unknown';

export const HOGE: HOGE_TYPE = 'hoge/HOGE_ACTION';
export const FUGA: FUGA_TYPE = 'fuga/FUGA_ACTION';"
`;

exports[`opts.filename 1`] = `
"// @flow
import type { HOGE_TYPE, FUGA_TYPE } from './actionTypes';

export const HOGE: HOGE_TYPE = 'hoge/HOGE_ACTION';
export const FUGA: FUGA_TYPE = 'fuga/FUGA_ACTION';"
`;

exports[`with extra 1`] = `
"// @flow
import type { HOGE_TYPE, FUGA_TYPE } from './unknown';

export const HOGE: HOGE_TYPE = 'hoge/HOGE_ACTION';
export const FUGA: FUGA_TYPE = 'fuga/FUGA_ACTION';"
`;
